src/ServerlessOpenApiDocumentation.ts

```diff
@@ -1,31 +1,58 @@
 import chalk from 'chalk';
 import * as fs from 'fs';
 import * as YAML from 'js-yaml';
+import _ = require('lodash');
+import * as Serverless from 'serverless';
 import { inspect } from 'util';
+
 import { DefinitionGenerator } from './DefinitionGenerator';
-import { IDefinitionType, ILog } from './types';
-import { merge } from './utils';
+import { Format, IDefinitionConfig, IDefinitionType, ILog } from './types';
+
+interface IOptions {
+  indent: number;
+  format: Format;
+  output: string;
+}
+
+interface IProcessedInput {
+  options: IOptions;
+}
+
+interface ICustomVars {
+  documentation: IDefinitionConfig;
+}
+
+interface IService {
+  custom: ICustomVars;
+}
+
+interface IVariables {
+  service: IService;
+}
+
+interface IFullServerless extends Serverless {
+  variables: IVariables;
+  processedInput: IProcessedInput;
+}
 
 export class ServerlessOpenApiDocumentation {
   public hooks;
   public commands;
   /** Serverless Instance */
-  private serverless;
-  /** CLI options */
-  // private options;
+  private serverless: IFullServerless;
+
   /** Serverless Service Custom vars */
-  private customVars;
+  private customVars: ICustomVars;
 
   /**
    * Constructor
    * @param serverless
    * @param options
    */
-  constructor (serverless, options) {
+  constructor (serverless: IFullServerless, options) {
```

Comment:
how come `options` doesn't need a type?

Reply:
This project is not configured(yet?) to use `strict` typescript transpilation.